analyses/cell-type-wilms-tumor-06/notebook/07_annotation_Across_Samples_exploration.Rmd

```diff
@@ -0,0 +1,586 @@
+     }
+    }
+  ) |>
+  # now combine all dataframes to make one big one
+  dplyr::bind_rows()
+
+```
+
+### Output file
+
+The report will be saved in the `notebook` directory. 
+
+
+## Functions
+
+#### do_Feature_mean
+
+`do_Feature_mean` shows heatmap of mean expression of a feature grouped by a metadata. 
+
+- `df` is the name of the table containing metadata and feature expression (counts) per cells
+- `group.by` is the name of the metadata to group the cells
+- `feature` is the name of the gene to average the expression
+
+```{r fig.width=10, fig.height=4, out.width='100%'}
+do_Feature_mean <- function(df, group.by, feature ){
+  
+  df <- df %>%
+  group_by(sample_id, !!sym(group.by))%>% 
+  summarise(m = mean(!!sym(feature)))  
+
+  
+p <- ggplot(df, aes(x= sample_id, y = !!sym(group.by), fill = m)) +
+   geom_tile() +
+   scale_fill_viridis_c()  +
+   theme_bw() +
+   theme(text = element_text(size = 20)) +
+   theme(axis.text.x = element_text(angle = 90, hjust = 0.5, vjust = 0.5)) +
+   guides(fill=guide_colourbar(title=paste0(feature)))
+
+return(p)
+}
+```
+
+
+#### do_Feature_boxplot
+
+`do_Feature_boxplot` shows boxplot of expression of a feature grouped by a metadata. 
+
+- `df` is the name of the table containing metadata and feature expression (counts) per cells
+- `group.by` is the name of the metadata to group the cells
+- `feature` is the name of the gene to average the expression
+
+```{r fig.width=10, fig.height=4, out.width='100%'}
+do_Feature_boxplot <- function(df, group.by, feature, split.by ){
+  
+  df <- df %>%
+    mutate(!!sym(group.by), group  = factor(!!sym(group.by), levels = c("fetal_nephron", "stroma", "immune", "endothelium")))
+    
+  p <- ggplot(df, 
+    aes(x = group, y = !!sym(feature), fill = group)
+  ) + 
+  geom_boxplot(size = 0.5, size.outlier = 0.25) +
+  facet_wrap(vars(!!sym(split.by)), scale = "free_y", ncol = 4) + 
+  theme_bw() +
+  theme(
+    axis.text.x = element_text(size = 12, angle = 0, hjust = 0.5, vjust = 0), 
+    legend.position = "none"
+  ) 
+
+return(p)
+}
+```
+
+## Analysis
+
+### Global cnv score
+
+As done in `06_cnv_infercnv_exploration.Rmd`, we calculate  single CNV score and assess its potential in identifying cells with CNV versus normal cells without CNV.
+
+We simply checked for each chromosome if the cell `has_cnv_chr`. 
+Would the cell have more than `cnv_threshold` chromosome with CNV, the global `has_cnv_score` will be TRUE.
+Else, the cell will have a `has_cnv_score` set to FALSE.
+
+
+
+
+```{r fig.width=10, fig.height=10, out.width='100%', results='asis'}
+
+#### CNV score
+  cell_type_df$has_cnv_score <- rowSums(cell_type_df[,grepl("has_cnv_chr", colnames(cell_type_df))])
+  cell_type_df$has_cnv_score[cell_type_df$has_cnv_score > params$cnv_threshold] <- "CNV"
+
+  # we know that some immune cells might have false positive CNV 
+  cell_type_df$has_cnv_score[cell_type_df$has_cnv_score %in% c(0:params$cnv_threshold)] <- "no CNV"
```

Comment:
regarding the `has_cnv_score`, I think we might want to allow the inference of up to 1 (or even 2) cnv in normal cells due to false positive. But this must be tested I think, this is why I decided to have the treshold as a parameter. 

Also the threshold might vary depending on the `infercnv` reference we used:
it seemed to me that using intra-patient normal cells might perform better in terms of false positive cnv than the inter-patient reference.

Reply:
Ah, ok I see the updated code: You are comparing to a threshold numerically, which makes sense. The updated code looks good to me! 
It would be good to also make a similar change to this in the `06` notebook as well to use `>` and `<=`, since `0:<number>` doesn't work unless the number itself is 0.